src/dc_tools.rs

```diff
@@ -1001,13 +997,22 @@ pub unsafe fn dc_get_fine_pathNfilename(
 }
 
 pub fn dc_is_blobdir_path(context: &Context, path: impl AsRef<str>) -> bool {
-    path.as_ref().starts_with(as_str(context.get_blobdir()))
+    context
+        .get_blobdir()
+        .to_str()
```

Comment:
any idea why `context.get_blobdir().starts_with(path)` does not work?

Reply:
because `Path` doesn't implement the `Pattern` trait, neede for `stars_with`

Reply:
Yet https://doc.rust-lang.org/std/path/struct.Path.html#method.starts_with is documented.

Mind you, I'm just trying to understand this not saying it needs changes - it's the bit I couldn't figure out when I was playing with this this morning.

Reply:
we are calling `starts_with` on `str` not on a path though